Various minor editorial improvements (3.0.4) (#3861)
OAI/OpenAPI-Specification: versions/3.0.4.md

The HTTP Status Codes are used to indicate the status of the executed operation. 
Status codes SHOULD be selected from the available status codes registered in the [IANA Status Code Registry](https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml).

> ##### <a name="httpAndCaseSensitivity"></a>HTTP and Case Sensitivity

Reviewer: ```suggestion
##### <a name="httpAndCaseSensitivity"></a>HTTP and Case Sensitivity

```

(some markdown renderers prefer a blank line after titles and most of the document does have them)

Author: I'm sure the only reason I didn't put one is that I saw one that didn't have them.  I prefer them set off by blank lines myself.

OAS uses several known formats to define in fine detail the data type being used.
However, to support documentation needs, the `format` property is an open `string`-valued property, and can have any value.
Formats such as `"email"`, `"uuid"`, and so on, MAY be used even though they are not defined by this specification.

> The OpenAPI Initiative also hosts a [Format Registry](https://spec.openapis.org/registry/format/) for formats defined by OAS users and other specifications.  Support for any registry format is strictly OPTIONAL, and support for one registry format does not imply support for any others.

Reviewer 2: I don't understand

> and support for one registry format does not imply support for any others

and would be less confused with ending the sentence after "OPTIONAL".

Reviewer 2: How about "registered format"? I was mislead to think about "format for the registry itself and its entries".

```suggestion
The OpenAPI Initiative also hosts a [Format Registry](https://spec.openapis.org/registry/format/) for formats defined by OAS users and other specifications.  Support for any registered format is strictly OPTIONAL, and support for one registered format does not imply support for any others.
```

### <a name="richText"></a>Rich Text Formatting
Throughout the specification `description` fields are noted as supporting CommonMark markdown formatting.

> Where OpenAPI tooling renders rich text it MUST support, at a minimum, markdown syntax as described by [CommonMark 0.27](https://spec.commonmark.org/0.27/). Tooling MAY choose to implement extensions on top of CommonMark 0.27, and MAY choose to ignore some CommonMark or extension features to address security concerns. 

Reviewer: I'm conflicted about expressly encouraging different tools to support different formats (conflicted because I work for a tools vendor that supports ridiculously good extensions that breaks other tooling if used!). The extensions were sort of already implied with "at a minimum", but this is a step further and I'm not sure it's helpful to users if they are expecting their API descriptions to be interoperable.

Author: @lornajane I only want to clarify the original intention.  I don't at all strongly about this, it's just one more issue to resolve.  If you and the TSC want to resolve the issue (#1867) without action I'm happy to take this commit out.

```

#### <a name="exampleObject"></a>Example Object

> An object grouping an internal or external example value with basic `summary` and `description` metadata.

Reviewer: ```suggestion

An object grouping an internal or external example value with basic `summary` and `description` metadata.
```

This object is typically used in properties named `examples` (plural), and is a [referenceable](#referenceObject) alternative to older `example` (singular) fields that do not support referencing or metadata.

> Allows sharing an example to demonstrate the usage of properties, parameters and objects within OpenAPI.

Reviewer: ```suggestion
Examples allow demonstration of the usage of properties, parameters and objects within OpenAPI.
```

(didn't like "sharing" in this context, tried to reword around it)
